src/ribasim_nl/ribasim_nl/model.py

```diff
@@ -92,6 +92,29 @@ def get_node(self, node_id: int):
         node_type = self.get_node_type(node_id)
         return getattr(self, pascal_to_snake_case(node_type))[node_id]
 
+    def remove_node(self, node_id: int, remove_edges: bool = False):
+        """Remove node from model"""
+        node_type = self.get_node_type(node_id)
+
+        # read existing table
+        table = getattr(self, pascal_to_snake_case(node_type))
+
+        # remove node from all tables
+        for attr in table.model_fields.keys():
+            df = getattr(table, attr).df
+            if df is not None:
+                getattr(table, attr).df = df[df.node_id != node_id]
+
+        if remove_edges and (self.edge.df is not None):
+            for row in self.edge.df[self.edge.df.from_node_id == node_id].itertuples():
+                self.remove_edge(
+                    from_node_id=row.from_node_id, to_node_id=row.to_node_id, remove_disconnected_nodes=False
+                )
+            for row in self.edge.df[self.edge.df.to_node_id == node_id].itertuples():
+                self.remove_edge(
+                    from_node_id=row.from_node_id, to_node_id=row.to_node_id, remove_disconnected_nodes=False
+                )
```

Comment:
Does this also remove the id from the Node table? And it support multiple data tables, like Basin / state and Basin / static?

Reply:
Yes, this code removes `node_id` from all tables found by `model_fields`, this include `Node`, `Static`, `State` (if basin), etc:
```
# remove node from all tables
for attr in table.model_fields.keys():
    df = getattr(table, attr).df
    if df is not None:
        getattr(table, attr).df = df[df.node_id != node_id]
```